awx/main/apps.py

```diff
@@ -63,3 +65,5 @@ def ready(self):
 
         self.load_credential_types_feature()
         self.load_named_url_feature()
+
+        discover_available_plugin_names()
```

Comment:
As I was looking into this, it occurred to me that an instance of this class might be available in places where the respective data is needed.
So perhaps it would make sense to set an attribute here. Something along the lines of:
```suggestion
        self.plugin_names = discover_available_plugin_names()
```
(provided that the function is updated to actually return something)

Or perhaps it could even be implemented via a `@property` (or, better yet, a `@cached_property`).

Reply:
It looks like there's a pattern of using `register()` above so it seems like that's what has to be used instead of putting the date onto an instance attribute.

Reply:
This has been resolved since the functionality has shifted to the serializer.